dns/dns_updater.go

```diff
@@ -97,7 +97,7 @@ func (u *updater) initELBs() error {
 
 	u.schemeToDNS = make(map[string]dnsDetails)
 	for scheme, lbDetails := range elbs {
-		u.schemeToDNS[scheme] = dnsDetails{dnsName: lbDetails.DNSName, hostedZoneID: lbDetails.HostedZoneID}
+		u.schemeToDNS[scheme] = dnsDetails{dnsName: lbDetails.DNSName + ".", hostedZoneID: lbDetails.HostedZoneID}
```

Comment:
Good catch. That looks like a bug to me, the ELB map should have a trailing dot as well. Otherwise it will always create an UPSERT for every single domain every time (safe, but inefficient). Can you add a
 check that `lbDetails.DNSName` doesn't have a trailing dot first, though? If it does (it shouldn't though), return an error.